janusgraph-test/src/main/java/org/janusgraph/graphdb/JanusGraphIndexTest.java

```diff
@@ -988,6 +993,60 @@ public void testRawQueries() {
         assertCount(numV / strs.length * 2, graph.indexQuery(PINDEX, "text:ducks").properties());
     }
 
+    /**
+     * Tests query parameters with raw indexQuery
+     */
+    @Test
+    public void testRawQueriesWithParameters() throws Exception {
+        if (!supportsLuceneStyleQueries()) return;
+        Parameter asc_sort_p = null;
+        Parameter desc_sort_p = null;
+
+        // ElasticSearch and Solr have different formats for sort parameters
+        String backend = config.get(ConfigElement.getPath(INDEX_BACKEND, INDEX), String.class);
+        if (backend == null) { // elasticsearch is default
+            Map<String,String> sortAsc = new HashMap<String,String>();
+            sortAsc.put("_score", "asc");
+            asc_sort_p = new Parameter("sort",Arrays.asList(sortAsc));
+            Map<String,String> sortDesc = new HashMap<String,String>();
+            sortDesc.put("_score", "desc");
+            desc_sort_p = new Parameter("sort",Arrays.asList(sortDesc));
+        } else if ("solr".equals(backend)) {
+            asc_sort_p = new Parameter("sort",new String[]{"score asc"});
+            desc_sort_p = new Parameter("sort",new String[]{"score desc"});
+        } else if ("lucene".equals(backend)) {
+            return; // Ignore for lucene
+        } else {
+            Assert.fail("Unknown index backend:" + backend);
+        }
+        
+        final PropertyKey field1Key = mgmt.makePropertyKey("field1").dataType(String.class).make();
+        mgmt.buildIndex("store1", Vertex.class).addKey(field1Key).buildMixedIndex(INDEX);
+        mgmt.commit();
+
+        JanusGraphVertex v1 = tx.addVertex();
+        JanusGraphVertex v2 = tx.addVertex();
+        JanusGraphVertex v3 = tx.addVertex();
+
+        v1.property("field1", "Hello Hello Hello Hello Hello Hello Hello Hello");
+        v2.property("field1", "Hello blue and yellow meet green");
+        v3.property("field1", "Hello");
+
+        tx.commit();
+
+        Thread.sleep(5000);
+
+        List<JanusGraphVertex> vertices = new ArrayList<JanusGraphVertex>();
+        for (JanusGraphIndexQuery.Result<JanusGraphVertex> r : graph.indexQuery("store1", "v.field1:(Hello)").addParameter(asc_sort_p).vertices()) {
+            vertices.add(r.getElement());
+        }
+        int idx=vertices.size()-1;
```

Comment:
Can we assert the size of the result set so that we are sure we got things back?